File: 0118.杨辉三角/0118-杨辉三角.rs
impl Solution {
    pub fn generate(n:i32) -> Vec<Vec<i32>> {
        
        let mut ret: Vec<Vec<i32>> = Vec::new();
        for i in 1..((n+1) as usize) {
            let mut arr = vec![1;i];
            for j in 1..i-1 {
                arr[j] = ret[i-2][j-1] + ret[i-2][j];
            }
            ret.push(arr);
        }
        return ret;
    }
}

File: 0118.杨辉三角/0118-杨辉三角.py
class Solution:
    def generate(self, n: int) -> List[List[int]]:
        if n == 0:
            return []
        ret = [[1]]
        for i in range(2, n + 1):
            tem = []
            last = ret[-1]
            tem.append(1)
            for j in range(1, i - 1):
                tem.append(last[j] + last[j - 1])
            tem.append(1)
            ret.append(tem)
        return ret
